### PR TITLE
http2: fix TODOs

### PR DESCRIPTION
**Pull request by Kriechi**
* http2: add support for too large data frames
* http2: add support for too large header frames
* http2: implement request target

### CODE REVIEW

**Comment by mhils** [comment]
That doesn't match https://httpwg.github.io/specs/rfc7540.html#rfc.section.8.1.2.3, does it?

**Comment by Kriechi** [comment]
PR updated - with an additional feature and a fix for your comment.
ping @mhils

File: netlib/http/http2/protocol.py [hunk 1]
```diff
@@ -80,13 +80,34 @@ def read_request(
...
+        scheme = headers.get_first(':scheme', 'https')
+        path = headers.get_first(':path', '/')
+        host = None
+        port = None
+
+        if path == '*' or path.startswith("/"):
+            form_in = "relative"
+        elif method == 'CONNECT':
+            form_in = "authority"
+            _, host, port, _ = utils.parse_url(scheme + "://" + authority)
```
Comment (mhils) [inline netlib/http/http2/protocol.py]: I think we can just split authority by `:` here?

**Comment by mhils** [comment]
LGTM :+1:

**Pull request merged**